main.go

```diff
@@ -330,6 +320,25 @@ func main() {
 	os.Exit(0)
 }
 
+func ensureIPMasqRule(config *subnet.Config, bn backend.Network) {
+	defer func() {
+		if err := network.TeardownIPMasq(config.Network, bn.Lease()); err != nil {
+			log.Errorf("Failed to tear down IP Masquerade: %v", err)
+		}
+	}()
+	// Repeatedly install the iptables rule (every second)
+	// in case it ever gets deleted
+	for {
+		err := network.SetupIPMasq(config.Network, bn.Lease())
+		if err != nil {
+			// If we fail to set up the IP Masquerade, don't keep retrying
+			log.Errorf("Failed to set up IP Masquerade: %v", err)
+			return
```

Comment:
not totally sure that this `return` is the right thing, it may be better to continue retrying instead

Reply:
Yes, I think that continuing to retry or returning an error in such a way that flanneld exits would be better.